_drafts/A BLE Primer.md

```diff
@@ -0,0 +1,1202 @@
+
+After that, the Connection is considered “created” but not yet “established”. A Connection is considered “established ” once the device receives a packet from its peer device. After a Connection becomes established, the Central becomes known as the Master, and the Peripheral becomes known as the Slave. The Master is responsible for managing the Connection, controlling the Connection Parameters and the timing of the different events within a Connection.
+
+#### Connection Events
+During a Connection Event, the Master and Slave alternate sending data packets to each other until neither side has data to send. Here are a few aspects of Connections that are very important  to know:
+
+- A Connection Event contains at least one packet sent by the Master.
+The Slave always sends a packet back if it received a packet from the Master.
+- If the Master does not receive a packet back from the Slave, the Master will close the Connection Event — it resumes sending packets at the next Connection Event.
+- The Connection Event can be closed by either side.
+- The starting points of consecutive Connection Events are spaced by a period of time called the Connection Interval.
+
+#### Connection Parameters
+The most important parameters that define a Connection include:
+
+- **Connection Interval**: the interval at which two connected BLE devices wake up the radio and exchange data (at each Connection Event).
+- **Slave Latency**: this value allows the Peripheral to skip a number of consecutive Connection Events and not listen to the Central at these Connection Events without compromising the Connection. For example, a Slave Latency of 3 allows a Slave to skip 3 Connection Events and a value of 0 means that the Slave has to send data to the Master at every Connection Event.
+- **Supervision Timeout**: the maximum time between two received data packets before the Connection is considered lost.
+
+## GATT, Services, and Characteristics
+The main reason to connect two BLE devices to each other is to transfer data between them. Without a Connection, it is not possible to have a bidirectional data transfer between two BLE devices.
+
+Which brings us to the concept of GATT...
+
+The Generic Attribute Profile (GATT) defines the format of the data exposed by a BLE device. It also defines the procedures needed to access the data exposed by a device.
+
+There are two Roles within GATT: Server and Client. The Server is the device that exposes the data it controls or contains, and possibly some other aspects of its behavior that other devices may be able to control. A Client, on the other hand, is the device that interfaces with the Server with the purpose of reading the Server’s exposed data and/or controlling the Server’s behavior.
+
+Keep in mind that a BLE device can act as the Server and a Client at the same time. Simply put, it acts as the Server for the sake of exposing its own data, and as a Client when accessing another device’s data.
```

Comment:
Keep in mind that a BLE device can act as **both** ~~the~~ **a** Server and a Client at the same time.

Reply:
Fixed